internal/services/azurestackhci/stack_hci_cluster_resource.go

```diff
@@ -145,6 +189,22 @@ func resourceArmStackHCIClusterRead(d *pluginsdk.ResourceData, meta interface{})
 		if props := model.Properties; props != nil {
 			d.Set("client_id", props.AadClientId)
 			d.Set("tenant_id", props.AadTenantId)
+
+			assignmentResp, err := hciAssignmentClient.Get(ctx, id.ResourceGroupName, id.ClusterName, "default")
+			if err != nil && !utils.ResponseWasNotFound(assignmentResp.Response) {
+				return err
+			}
+
+			if !utils.ResponseWasNotFound(assignmentResp.Response) && assignmentResp.Properties != nil && assignmentResp.Properties.ConfigurationProfile != nil {
+				automanageConfigId, err := autoParse.AutomanageConfigurationID(*assignmentResp.Properties.ConfigurationProfile)
+				if err != nil {
+					return err
+				}
+				d.Set("automanage_configuration_id", automanageConfigId.ID())
+			} else {
+				// remove the automanage_configuration_id if it is not found
+				d.Set("automanage_configuration_id", "")
+			}
```

Comment:
Could you refactor this to
```suggestion
			configId := ""
			if !utils.ResponseWasNotFound(assignmentResp.Response) && assignmentResp.Properties != nil && assignmentResp.Properties.ConfigurationProfile != nil {
				automanageConfigId, err := autoParse.AutomanageConfigurationID(*assignmentResp.Properties.ConfigurationProfile)
				if err != nil {
					return err
				}
				configId = automanageConfigId.ID()
			} 
			
			d.Set("automanage_configuration_id", configId)
```